### PR TITLE
Front end mutation cleanup

### PR DESCRIPTION
**Pull request by wongcmichelle**
[Notion ticket](https://www.notion.so/uwblueprintexecs/delete-and-edit-mutation-cleanup-98ad81733016484b928909fb184e7360)

## Changes
For each entity type
- Move url navigation after add/edit to `useMutation`'s `onCompleted` callback
    - Use `history.replace()` instead of `history.push()`
- Move `enqueueSnackbar` calls to `useMutation`'s `onCompleted` callback
- Add missing snackbars (based on Figma)

Entity types
- Ambulance
- CCP
- Event
- Hospital
- Patient
- User

Additional scope (added to notion ticket):
- Remove option menu when the delete confirmation modal opens (should be fixed for the following)
    - Ambulance
    - CCP
    - Hospital
    - User

## Testing
For each of the listed entities above:
- Check add/edit/delete mutations
    - Confirm cache/UI updates correctly
    - Snackbars appear (if in Figma designs)
- Check the option menu is closed when the delete confirmation modal opens

Here are the urls for accessing the general flow for mutating each entity:
- Ambulance: `/manage/ambulances`
- CCP: `/events/{ccpId}`
- Event: `/events`
- Hospital: `/manage/hospitals`
- Patient: `/events/{eventId}/ccps/{ccpId}`
- User (also referred to as team member): `/manage/members`

## Screenshots
Ambulance
image

CCP
image

Hospital
image

User
image

before opening PR
- [x] check notion ticket
- [x] run linter
- [x] go through file diff

filling out PR
- [x] descriptive title
- [x] update `notion ticket` link
- [x] fill out `Changes`
  - [x] \(optional) add inline comments in diff
- [x] fill out `Testing`
- [x] \(optional) add `Screenshots`
- [x] assign yourself to the PR

after opening PR
- [x] link PR to notion ticket
- [x] move ticket to `In PR`
- [ ] make sure build passes
- [x] ping `@ps` in `#paramedics-dev`

### CODE REVIEW

File: src/components/EventDashboard/CCPTabPanel.tsx [hunk 1]
```diff
@@ -159,7 +159,7 @@ const CCPTabPanel = ({ eventId }: { eventId: string }) => {
       });
 
       const updatedCCPsList = collectionPointsByEvent.filter(
-        (ccp) => ccp.id !== (selectedCCP as CCP).id
+        (ccp) => ccp.id !== deleteCollectionPoint
```
Comment (wongcmichelle) [inline src/components/EventDashboard/CCPTabPanel.tsx:162]: The backend has been updated to return the `id` of the deleted entity, so we're updating this to check using that returned `id`.

File: src/components/EventOverview/EventsPage.tsx [hunk 2]
```diff
@@ -112,7 +111,6 @@ const EventsPage = () => {
   };
 
   const handleDeleteEvent = async (event: Event) => {
```
Comment (petorsong) [inline src/components/EventOverview/EventsPage.tsx]: not sure if this is really within the scope but is this `async` here pointless?

**Review by petorsong — CHANGES_REQUESTED** [review]
getting an error when attempting to delete ambulances
image

weirdness around attempting to delete a CCP - `delete` button on modal cannot be clicked and the modal goes haywire after
Paramedics React - Mozilla Firefox 2020-10-13 21-52-04

**Review by aaronabraham311 — CHANGES_REQUESTED** [review]
Couple of things that I noticed:
1. I cannot delete a CCP. When I try to delete a CCP, nothing changes (attached a GIF with my issue). Checking the network console shows that no requests were made.
Kapture 2020-10-16 at 21 14 14
2. Getting the same haywire issues as Peter after I try to delete a CCP.
3. No snack bars are appearing for patient add, edit or delete
4. No snack bars are appearing for add, edit or delete events

**Comment by wongcmichelle** [comment]
> Couple of things that I noticed:
> 
> 1. I cannot delete a CCP. When I try to delete a CCP, nothing changes (attached a GIF with my issue). Checking the network console shows that no requests were made.
>    Kapture 2020-10-16 at 21 14 14
> 2. Getting the same haywire issues as Peter after I try to delete a CCP.
> 3. No snack bars are appearing for patient add, edit or delete
> 4. No snack bars are appearing for add, edit or delete events

@aaronabraham311  and I were unable to reproduce this bug. (1 & 2)
For 3 & 4, this is expected as snackbars are not included for the listed patient and event mutations in Figma.

File: src/components/EventOverview/EventsPage.tsx (hunk 2 above)
Comment (petorsong) [reply src/components/EventOverview/EventsPage.tsx]: bump this just in case @wongcmichelle

**Review by petorsong — DISMISSED** [review]
super weird - can't repro the previous issues on my end either

**Comment by aaronabraham311** [comment]
I'm not sure if we want to tackle this on this ticket but I checked Figma and the snack bars are included for patients (eg. patient edit: https://www.figma.com/file/IpOPyv79gBwEohvOJvPV6f/%F0%9F%9A%91-Designs-for-Development-S20-F20?node-id=2020%3A1275)

**Comment by wongcmichelle** [comment]
> I'm not sure if we want to tackle this on this ticket but I checked Figma and the snack bars are included for patients (eg. patient edit: https://www.figma.com/file/IpOPyv79gBwEohvOJvPV6f/%F0%9F%9A%91-Designs-for-Development-S20-F20?node-id=2020%3A1275)

That is a separate task :) cc @petorsong

**Comment by petorsong** [comment]
> 
> 
> > I'm not sure if we want to tackle this on this ticket but I checked Figma and the snack bars are included for patients (eg. patient edit: https://www.figma.com/file/IpOPyv79gBwEohvOJvPV6f/%F0%9F%9A%91-Designs-for-Development-S20-F20?node-id=2020%3A1275)
> 
> That is a separate task :) cc @petorsong

yep it'll be handled here https://github.com/uwblueprint/paramedics-react/pull/39

File: src/components/EventOverview/EventsPage.tsx (hunk 2 above)
Comment (wongcmichelle) [reply src/components/EventOverview/EventsPage.tsx]: Good catch, thanks :)

**Review by aaronabraham311 — APPROVED** [review]

**Review by petorsong — APPROVED** [review]

**Pull request merged**